src/main/java/org/junit/internal/builders/AllDefaultPossibilitiesBuilder.java

```diff
@@ -40,7 +40,10 @@ protected JUnit3Builder junit3Builder() {
     }
 
     protected AnnotatedBuilder annotatedBuilder() {
-        return new AnnotatedBuilder(this);
+        // For compatibility with Request#classes(...) and the Suite runner,
+        // deeper nesting levels always need to support suite() methods:
+        RunnerBuilder suiteBuilder = canUseSuiteMethod ? this : new AllDefaultPossibilitiesBuilder(true);
```

Comment:
AllDefaultPossibilitiesBuilder is clearly an internal class without any API, so nobody playing by the rules can be broken by my proposed fix.

There is no use case for the current behavior of `new AllDefaultPossibilitiesBuilder(false)`. It's just an implementation bug that should be fixed in JUnit 4. The Javadoc of Request#classWithoutSuiteMethod() supports that view by saying that it "will run all the tests in a class. If the class has a suite() method, it will be ignored."
=> Only the suite method of the given class should be ignored.

The only reason why that special request exists is to avoid the endless nesting that would otherwise occur if a test class uses the JUnit4TestAdapter in its suite() method (according to the API of JUnit4TestAdapter, see the added Javadoc in my fix).